admin news feature

vkorir opened the pull request:
1. Login into wise as an admin user
2. Open the menu and select "Admin Tools"
3. Click the "Manage News" card button
4. Verify that you can see the Add/Edit/Hide/Delete buttons as an admin
5. Try creating/editing/deleting news items and verify this works as expected
6. Verify that the search bar on top filters the news items accordingly

closes #2216

vkorir commented:
I wanted to clarify the purpose of "Read More" button, I saw it on @breity 's mockups and wasn't sure what it should do since we're already showing the whole news item without hiding any parts.

breity commented:
> I wanted to clarify the purpose of "Read More" button, I saw it on @breity 's mockups and wasn't sure what it should do since we're already showing the whole news item without hiding any parts.

@vkorir Hey Victor. Sorry, that's supposed to be an "Edit" button on the admin page, not "Read More". I updated it here: https://balsamiq.cloud/s8brvqj/p9j1not/r644E.

vkorir commented:
@geoffreykwan I'm having trouble cherry-picking commits from the admin search [PR](https://github.com/WISE-Community/WISE/pull/2008). I think it's because I'm using part of the code that I introduced when implementing the admin search feature. however, I addressed all your comments and updated this PR.

geoffreykwan commented:
The changes look good. Here are couple more minor things.

1. When creating or editing a news item can you get rid of the "Teacher Only" option. We've decided we don't need that anymore.

2. For the hide/unhide confirmation messages can you change them a bit

Change
```Are you sure you want to show this news item? This news item will be visible to everyone.```
to
```Are you sure you want to show this news item to users?```

Change
```Are you sure you want to hide this news item? This news item will nolonger be visible to non-admin users.```
to
```Are you sure you want to hide this news item from users?```

3. Use i18n with the messages in news-item-dialog.component.ts so they can be translated in the future. This includes the hide/unhide messages as well as the snackbar messages. You can look at share-run-dialog.component.ts to see an example of how we use i18n in components.

4. As for the cherry-picking, I'll handle this after everything has been completed in this task. We've made a lot of changes on the develop branch so merging it in might be messy.
